app/models/admin_dashboard_data.rb

```diff
@@ -3,17 +3,48 @@
 class AdminDashboardData
   include StatsCacheable
 
+  class Problem
+    attr_reader :message, :priority, :identifier
+
+    def initialize(message, priority: "low", identifier: nil)
+      @message = message
+      @priority = if !["low", "high"].include?(priority)
+        "low"
```

Comment:
Very minor, but maybe we can extract VALID_PRIORITIES constant

Also, because we have if/else block we can remove negation. Something like
if ["low", "high"].include?(priority)
  priority
else
  "low"
end